workbench-core/datasets/src/storageLocation.ts

```diff
@@ -0,0 +1,11 @@
+/*
+ *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *  SPDX-License-Identifier: Apache-2.0
+ */
+
+export interface StorageLocation {
+  name: string;
+  awsAccountId?: string;
+  type?: string;
+  region?: string;
```

Comment:
Why are `awsAccountId`, `type` and `region` optional?

Reply:
`awsAccountId` and `region` are AWS specific parameters, and since DatasetsService isnt bound to AWS keeping them optional makes sense. `type` should be required, I will update that.